Core/Types/CkanModule.cs

```diff
@@ -623,6 +623,22 @@ public string DescribeInstallStanzas()
             return string.Join(", ", descriptions);
         }
 
+        /// <summary>
+        /// Return an archive.org URL for this download, or null if it's not there.
+        /// The filenames look a lot like the filenames in Net.Cache, but don't be fooled!
+        /// Here it's the first 8 characters of the SHA1 of the DOWNLOADED FILE, not the URL!
+        /// </summary>
+        public Uri InternetArchiveDownload
+        {
+            get
+            {
+                return license.All(l => l.Redistributable)
+                    ? new Uri(
+                        $"https://archive.org/download/{identifier}-{version}/{download_hash.sha1.Substring(0, 8)}-{identifier}-{version}.zip")
+                    : null;
+            }
+        }
+
         /// <summary>
```

Comment:
Some tests of the filename generation could be useful, but I wouldn't treat this as a blocker.